internal/vehicle/ford.go

```diff
@@ -134,36 +170,139 @@ func (v *Ford) vehicles() ([]string, error) {
 	return vehicles, err
 }
 
-// chargeState implements the api.Vehicle interface
-func (v *Ford) chargeState() (float64, error) {
-	var resp struct {
-		VehicleStatus struct {
-			Battery struct {
-				BatteryStatusActual struct {
-					Value int
-				}
+// Parses vehicle timestamp in format returned by Ford API and calculates age
+func (v *Ford) CalculateAge(timestamp string) (age time.Duration, err error) {
+	var timestampTime time.Time
+	const dateFormat = "01-02-2006 15:04:05"
+	timestampTime, err = time.Parse(dateFormat, timestamp)
+	age = time.Since(timestampTime)
+
+	return age, err
+}
+
+// VehicleStatus implements the /status response
+func (v *Ford) VehicleStatus() (res vehicleStatus, err error) {
+	uri := fmt.Sprintf("%s/api/vehicles/v3/%s/status", fordAPI, v.vin)
+
+	var req *http.Request
+	req, err = v.request(http.MethodGet, uri)
+	if err == nil {
+		err = v.DoJSON(req, &res)
+	}
+
+	var statusAge time.Duration
+	if err == nil {
+		statusAge, err = v.CalculateAge(res.VehicleStatus.LastRefresh)
+	}
+
+	if err == nil && statusAge > outdatedAfterMins*time.Minute {
+		// received data is considered outdated, server is requested to poll updated data from vehicle
+		v.log.DEBUG.Printf("Vehicle Status is outdated (age %v > %dm), requesting refresh", statusAge, outdatedAfterMins)
+		var updatedRes vehicleStatus
+		updatedRes, err = v.VehicleStatusRefresh()
+		if err == nil {
+			res = updatedRes
+			statusAge, err = v.CalculateAge(res.VehicleStatus.LastRefresh)
+			v.log.DEBUG.Printf("Refreshed Status Age: %v", statusAge)
+		}
+	}
+
+	return res, err
+}
+
+// Get updated vehicle status after requesting refresh
+func (v *Ford) VehicleStatusRefresh() (res vehicleStatus, err error) {
+	var commandId string
+	commandId, err = v.requestRefresh()
+
+	if err == nil {
+		uri := fmt.Sprintf("%s/api/vehicles/v3/%s/statusrefresh/%s", fordAPI, v.vin, commandId)
+
+		var req *http.Request
+
+		counter := 0
+		for counter < maxTrials {
```

Comment:
Wie wärs mit

    for err == nil && res.Status != http.StatusOK && counter++ < maxTrials {

Reply:
Spart Dir beide Breakstatements

Reply:
just as a remark: the API always responds with HTTP status 200, the relevant status is hidden in the status attribute of the JSON data; as res seems to be initialized with Status = 0, your suggestion still works. I had to slightly modify it to pass syntax check:
`for counter := 0; counter < fordMaxRefreshTrials && res.Status != 200 && err == nil; counter++ {`